__tests__/endpoints/app.test.js

```diff
@@ -34,3 +35,29 @@ describe("GET /api/topics", () => {
       });
   });
 });
+
+describe("GET /api", () => {
+  test("200 and return an object with all the endpoints available, with their descriptions", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        for (let endpoint in body) {
+          const apiEndpoint = body[endpoint];
+          expect(typeof apiEndpoint).toBe("object");
+          expect(typeof endpoint).toBe("string");
+          expect(typeof apiEndpoint.description).toBe("string");
+          expect(apiEndpoint).toMatchObject(endpointsData[endpoint]);
+          if (apiEndpoint.queries) {
+            expect(Array.isArray(apiEndpoint.queries)).toBe(true);
+          }
+          if (apiEndpoint.exampleResponse) {
+            expect(typeof apiEndpoint.exampleResponse).toBe("object");
+          }
+        }
+        expect(Object.keys(body).length).toBe(
+          Object.keys(endpointsData).length
+        );
+      });
```

Comment:
The idea behind this test is to ensure what we get back as a response from `/api` is the file contents of `endpoints.json`, so we dont actually need to do anything this complex to test for this. Remember you can require in json files and access the objects directly which means all this test really needs is for us to assert that the body from this response looks the same as the required in json object from endpoints.json.